packages/protocol/contracts/authorizers/Api3Authorizer.sol

```diff
@@ -125,7 +130,7 @@ contract Api3Authorizer is IApi3Authorizer {
   ) external view override returns (bool) {
     return
       designatedWallet.balance != 0 &&
-      !clientAddressToBlacklistStatus[clientAddress] &&
-      airnodeIdToClientAddressToWhitelistExpiration[airnodeId][clientAddress] > block.timestamp;
+      (airnodeIdToClientAddressToWhitelistStatus[airnodeId][clientAddress] ||
+        airnodeIdToClientAddressToWhitelistExpiration[airnodeId][clientAddress] > block.timestamp);
```

Comment:
An `airnodeId`-`clientAddress` pair is authorized if its status is true or its expiration is in the future. I know this is a bit counterintuitive but this is the functionality we want and we can potentially work around that by some remaining.
Say the integrations team has an admin (not super-admin) wallet and set the whitelist expiration time as next month. Authorizer should return true. During that, the user locked some tokens and the token lock contract, which is a super-admin, set the whitelist status to be true. Then, the authorizer should still remain true. Then, the user unlocked their tokens and the token lock contract set their whitelist status as false. Since their whitelisting set by the authorizer team hasn't still expired, the authorizer should still remain true. So this is the main reason we're keeping the status and the expiration separately and we need to give access if either of them satisfies the condition.

Reply:
Got it, updated the tests to reflect this